extensions/Worker.Extensions.CosmosDB/src/Config/CosmosDBBindingOptions.cs

```diff
@@ -21,6 +21,10 @@ internal class CosmosDBBindingOptions
 
```

Comment:
I'm setting the options like this:

```csharp
builder.Services.Configure<CosmosDBOptions>((cosmosOptions) =>
{
    cosmosOptions.ConnectionMode = ConnectionMode.Direct;
    cosmosOptions.UserAgentSuffix = "MyApp";
});
```

Seeing as null in logs:

> [2024-05-09T20:28:49.395Z] CosmosDBOptions
[2024-05-09T20:28:49.395Z] {
[2024-05-09T20:28:49.395Z]   "ConnectionMode": null
[2024-05-09T20:28:49.395Z] }

But the ConnectionMode is being set properly when I log it as part of the function code:

`_logger.LogInformation("Cosmos client options: {0} - {1}", client.ClientOptions.ConnectionMode, client.ClientOptions.ApplicationName);`

> Cosmos client options: Direct - MyApp

Any thoughts on why it's null when logged during startup?

Reply:
I'm not logging it myself, it's just part of the host start up logs? I'm not actually sure what does this i.e.

```log
[2024-05-09T20:28:49.392Z] FunctionResultAggregatorOptions
[2024-05-09T20:28:49.393Z] {
[2024-05-09T20:28:49.393Z]   "BatchSize": 1000,
[2024-05-09T20:28:49.393Z]   "FlushTimeout": "00:00:30",
[2024-05-09T20:28:49.393Z]   "IsEnabled": true
[2024-05-09T20:28:49.393Z] }
[2024-05-09T20:28:49.395Z] SingletonOptions
[2024-05-09T20:28:49.395Z] {
[2024-05-09T20:28:49.395Z]   "LockPeriod": "00:00:15",
[2024-05-09T20:28:49.395Z]   "ListenerLockPeriod": "00:00:15",
[2024-05-09T20:28:49.395Z]   "LockAcquisitionTimeout": "10675199.02:48:05.4775807",
[2024-05-09T20:28:49.395Z]   "LockAcquisitionPollingInterval": "00:00:05",
[2024-05-09T20:28:49.395Z]   "ListenerLockRecoveryPollingInterval": "00:01:00"
[2024-05-09T20:28:49.395Z] }
[2024-05-09T20:28:49.395Z] CosmosDBOptions
[2024-05-09T20:28:49.395Z] {
[2024-05-09T20:28:49.395Z]   "ConnectionMode": null
[2024-05-09T20:28:49.395Z] }
[2024-05-09T20:28:49.395Z] ScaleOptions
[2024-05-09T20:28:49.395Z] {
[2024-05-09T20:28:49.396Z]   "ScaleMetricsMaxAge": "00:02:00",
[2024-05-09T20:28:49.396Z]   "ScaleMetricsSampleInterval": "00:00:10",
[2024-05-09T20:28:49.396Z]   "MetricsPurgeEnabled": true,
[2024-05-09T20:28:49.396Z]   "IsTargetScalingEnabled": true,
[2024-05-09T20:28:49.396Z]   "IsRuntimeScalingEnabled": false
[2024-05-09T20:28:49.396Z] }
[2024-05-09T20:28:49.398Z] Starting JobHost
```

Reply:
The host automatically logs the options that implement `IOptionsFormatter` at initialization to show how those options were setup *for the extension*. So anything set on the worker will not have any impact on the host/extension side.